tools/clang/include/clang/SPIRV/SpirvInstruction.h

```diff
@@ -470,6 +470,11 @@ class SpirvDecoration : public SpirvInstruction {
                   spv::Decoration decor,
                   llvm::ArrayRef<SpirvInstruction *> params);
 
+  SpirvDecoration(SourceLocation loc, SpirvInstruction *target,
+                  SpirvFunction *targetFunc, spv::Decoration decor,
+                  llvm::ArrayRef<uint32_t> params = {},
+                  llvm::Optional<uint32_t> index = llvm::None);
```

Comment:
Since we have
```
  SpirvDecoration(SourceLocation loc, SpirvInstruction *target,
                  spv::Decoration decor,
                  llvm::ArrayRef<SpirvInstruction *> params);
```

I think the new one can be simply
```
  SpirvDecoration(SourceLocation loc, SpirvFunction *targetFunc, spv::Decoration decor,
                  llvm::ArrayRef<uint32_t> params = {},
                  llvm::Optional<uint32_t> index = llvm::None);
```

In addition, just out of curiosity, do we have a specific case that we need `llvm::Optional<uint32_t> index = llvm::None`?

I am not sure SPIR-V `OpMemberDecorate` has a function operand.
Could you please remove `llvm::Optional<uint32_t> index = llvm::None` parameter if you do not have a specific example?
We can add it later when we actually meet the case.